### ISSUE TITLE
Add tests for function arguments

### ISSUE THREAD

**Issue by alifeee** [burnash/gspread#1303]
Some functions are aliases of other functions. For example:

- `get_all_values` and `get_values`
- `get_all_records` and `get_records`

Over time, we may add kwargs and args to the original functions, and forget to add these to the aliases. We should add some simple tests to the test files that verify that the function arguments/kwargs are the same.

This issue alread arose once with `get` vs `get_values`.

**Comment by alifeee** [comment]
I don't know that this is possible for functions that use the [`@accepted_kwargs`][accepted kwargs] decorator

This may be easier in v6.0.0. Arguments and kwargs should be explicitly set, and the length can be obtained via the signature. See this SO post:

<https://stackoverflow.com/a/41188411>

```python
from inspect import signature

def someMethod(self, arg1, kwarg1=None):
    pass

sig = signature(someMethod)

print(len(params))  # 3
```

[accepted kwargs]: https://github.com/burnash/gspread/blob/7fe63bf9e2b8252db914cc47b9a39f4fabc57019/gspread/utils.py#L685-L738

**Comment by lavigne958** [comment]
The functions/methods using `accepted_kwargs` will be hard to update. We introduced a new special value to allow arguments with no value. Apart from that we can't do much.

The target is in 6.0.0 we removed the `accepted_kwargs` decorator so the truth is in the arguments themselves and their value too. And we will try to reduce the number of occurrences of `**kwargs` to make it explicit. 

We can put it as a target:
- remove all ocurences of `**kwargs` 


What do you think?

**Comment by alifeee** [comment]
I agree, we should not have

```python
def func(**kwargs):
...
```

and we *should* have

```python
def func(kw_arg1=True, kw_arg2=False, ...)
```

I would like to add a simple test that `get` and `get_values` and `get_all_values` keep the same number of kwargs, for example (if/when we add new formatting options, they should be available to all functions)

This is ultimately low priority.

**Comment by alifeee** [comment]
closed by #1296
